src/domain/flow-system/domain/src/entities/flow/flow_id.rs

```diff
@@ -8,17 +8,23 @@
 // by the Apache License, Version 2.0.
 
 use internal_error::InternalError;
+use serde::{Deserialize, Serialize};
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
 /// Uniquely identifies a flow
-#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
-pub struct FlowID(u64);
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash, Serialize, Deserialize)]
+pub struct FlowID(i64);
```

Comment:
Databases are very strict with sequence types

Reply:
Would it be better to add casts in DB implementations instead?

I think we should not let infra dictate us the domain types - domain types should make invalid values like negative IDs unrepresentable.

There's a similar situation in ODF where Java is notoriously bad at `unisigned` types, but if we relax types in the domain just because some clients don't like unsigned types this causes a huge wave of negative effects.

This is probably not the case here though, as event IDs are opaque and likely won't ever be manipulated on.

Reply:
Okay, addressed. I've added some conversion traits for convenience